doubles/allowance.py

```diff
@@ -1,6 +1,8 @@
 from functools import wraps
 from inspect import getargspec
 
+from concurrent.futures import Future
```

Comment:
This import should probably go inside the functions below so we don't break doubles entirely when futures is are not supported.